### PR TITLE
Add pull request (PR) template

### PR DESCRIPTION
**Pull request by josh-wong**
## Description

This PR updates the PR template in this repository. The template is for helping to provide context for people who are involved with PRs and for future reference.

## Related issues and/or PRs

N/A

## Changes made

Created the pull request template in the **.github** folder.

## How these changes were tested

Checked that this PR template looks as expected in GitHub.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

**Review by supl — APPROVED** [review]
Thank you, Josh 🙇

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Description
+
+> Provide a brief description about **why** this PR is necessary. Be sure to provide context.
+
+## Related issues and/or PRs
+
+> If this PR addresses or references any issues and/or other PRs, list them here. For example, "Fixes #123", "Related to #456", or "Addresses #789".
```
Comment (kota2and3kan) [inline .github/pull_request_template.md]: GitHub has a feature [Linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that is:

* Link PR and issue automatically by using specific keywords.
* If PR is closed, GitHub automatically closes the linked issues.

Regarding this feature, `Fixes #123` is one of the keywords (you can see the list of keywords in the above linked document). So, if we use `Fixes`, GitHub links and closes related issues automatically.

However, `Related to #456` and `Addresses #789` are not keywords of this feature. So, if we use these keywords, GitHub does not link and close related issues automatically.

In other words, the behavior will be different depending on `which word we will use`. And, there is no description of this GitHub's linking feature in this template.

So, I think it causes a bit of confusion.

I think it would be better to:

* Describe (add link) the GitHub's linking feature in this template.
* Do NOT use `Fixes` that invoke the GitHubs' linking feature.
* Use other keywords instead of `Related to` or `Address`, to invoke GitHub's linking feature always.

What do you think?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- [ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
+- [ ] Tests (unit, integration, etc.) have been added for the changes.
+- [ ] My changes generate no new warnings.
+- [ ] Any dependent changes in other PRs have been merged and published.
+
+## Additional notes (optional)
+
+> Provide any additional information or notes that may be relevant to the reviewers or stakeholders.
+
+## Release notes
```
Comment (kota2and3kan) [inline .github/pull_request_template.md]: Just a question.

We don't have any release notes for this repository (for in-house docker image), but do we need to this section in the PR template in this repository?

Or, do you have any plan to create a release note for this in-house docker image in the future?

**Review by kota2and3kan — COMMENTED** [review]
Thank you for creating the PR template!
I left one suggestion and one question.
Please take a look when you have time!

File: .github/pull_request_template.md (hunk 1 above)
Comment (josh-wong) [reply .github/pull_request_template.md]: Thank you for sharing this. I didn't know that GitHub had this feature.

What do you think about simplifying this description as follows?:

```suggestion
## Related issues and/or PRs

> If this PR addresses or references any issues and/or other PRs, list them here.
```

File: .github/pull_request_template.md (hunk 2 above)
Comment (josh-wong) [reply .github/pull_request_template.md]: Since this repo is in-house only, I'm OK with removing the `## Release note` section. If we decide that we need the section later, we can always re-add it. Is that OK with you?

File: .github/pull_request_template.md (hunk 1 above)
Comment (kota2and3kan) [reply .github/pull_request_template.md]: > What do you think about simplifying this description as follows?

Thank you for the update!
This simplified description looks good to me!

File: .github/pull_request_template.md (hunk 2 above)
Comment (kota2and3kan) [reply .github/pull_request_template.md]: Yes. Removing the `## Release note` section is OK with me! It sounds good!

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!
